clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -4707,6 +4711,35 @@ void SYCLIntegrationHeader::emit(raw_ostream &O) {
     O << "  __SYCL_DLL_LOCAL\n";
     O << "  static constexpr bool isESIMD() { return " << K.IsESIMDKernel
       << "; }\n";
+    O << "  __SYCL_DLL_LOCAL\n";
+    O << "  static constexpr const char* getFileName() { return "
+#ifndef NDEBUG
+      << std::string(PLoc.getFilename())
+             .substr(std::string(PLoc.getFilename()).find_last_of("/\\") + 1)
+#endif
+      << "; }\n";
+    O << "  __SYCL_DLL_LOCAL\n";
+    O << "  static constexpr const char* getFunctionName() { return "
+#ifndef NDEBUG
+      << K.NameType->getAsCXXRecordDecl()->getName()
```

Comment:
Alexey commented on the #ifndef lines, you absolutely don't want those in SemaSYCL.cpp (they are to be printed presumably?).

Is this really sufficient to meet the requirements to just print the provided name?  I would expect the users would want a fully-qualified name.

Reply:
> Alexey commented on the #ifndef lines, you absolutely don't want those in SemaSYCL.cpp (they are to be printed presumably?).

@alexbatashev can you clarify the requirements here? Do you want different output generated in integration header based on whether release/debug build of compiler is used?

Reply:
No, the requirement here is to disable debug info, when user compiles their application in release mode. So, we really need these #ifdefs in the integration header, not in the compiler itself.

Reply:
@alexbatashev Such as in the suggestion here? https://github.com/intel/llvm/pull/5335#discussion_r795956472

Reply:
> Alexey commented on the #ifndef lines, you absolutely don't want those in SemaSYCL.cpp (they are to be printed presumably?).
I interpreted this comment to mean that @erichkeane was against what @alexbatashev said here: #5335 (comment) My bad. It seems you both agreed?

Reply:
yep, that was me agreeing with Alexey.

Reply:
https://github.com/intel/llvm/pull/5335/commits/16d6af3670f802e0ec44f662baaf6aed5d1f370a